community/security/src/main/java/org/neo4j/server/security/auth/BasicAuthManager.java

```diff
@@ -173,7 +169,7 @@ public void setUserPassword( String username, String password ) throws IOExcepti
 
         if ( existingUser.credentials().matchesPassword( password ) )
```

Comment:
The password policy functionality is duplicated in REST UserService, with its own expected status code, hence the division of a setUserPassword (it is the old API renamed)
Here is an attempt to remove password policy from UserService and move validation into setUserPassword (but it changes the expected status code of one test..): https://github.com/henriknyman/neo4j/commit/4d79ced5def19fd9ea9ac599d050b371b2fa3f32